allennlp/data/tokenizers/pretrained_transformer_tokenizer.py

```diff
@@ -109,13 +82,26 @@ def tokenize(self, text: str) -> List[Token]:
             truncation_strategy=self._truncation_strategy,
             return_tensors=None,
         )
-        # token_ids containes final list with ids for both regualr and special tokens
-        token_ids = encoded_tokens["input_ids"]
-        token_type_ids = encoded_tokens["token_type_ids"]
-
-        token_strings = self._tokenizer.convert_ids_to_tokens(token_ids, skip_special_tokens=False)
+        # token_ids containes a final list with ids for both regualr and special tokens
+        token_ids, token_type_ids = encoded_tokens["input_ids"], encoded_tokens["token_type_ids"]
 
         tokens = []
-        for token_str, token_id, token_type_id in zip(token_strings, token_ids, token_type_ids):
+        for token_id, token_type_id in zip(token_ids, token_type_ids):
+            token_str = self._tokenizer.convert_ids_to_tokens(token_id, skip_special_tokens=False)
             tokens.append(Token(text=token_str, text_id=token_id, type_id=token_type_id))
+
         return tokens
+
+    def tokenize_sentence_pair(self, sentence_1: str, sentence_2: str):
```

Comment:
I was suggesting putting this on the base class, but I suppose putting it here for now is fine, and we can add it higher up later if necessary.  This just means that the dataset reader has to instantiate a `PretrainedTransformerTokenizer`, instead of just a `Tokenizer`, for the types to work out.

Reply:
I added the commit representing this approach.